### PR TITLE
Improve architecture

### LINKED ISSUE #59: Fix CI to be more elegant
lint should use yarn lint, lint should run on prs

Also use github cache!!!!

### LINKED ISSUE #67: Add a CI/CD Cache
Nice to have to mitigate build times

### LINKED ISSUE #68: Add script to generate JWT token
Add in a package.json script to generate JWT tokens!

### LINKED ISSUE #74: Fix eslint & remove .editorconfig
editorconfig doesn't format very well in terms of strict styling, and eslint currently isn't configured properly

### PR DESCRIPTION
**Pull request by Timothy-Gonzalez**
A bunch of QOL changes and fixes a whole lot of stuff to do with architecture.

In summary:
- [x] Remove .editorconfig (Resolves #74)
- [x] Added prettier to format (Resolves #74)
- [x] Changed eslint to only be a linter (Resolves #74)
- [x] Fixed vercel structure to be a little more elegant
- [x] Remove coupled dependency from vercel (build & run are detached from vercel now)
- [x] Upgraded from CJS to ESM
- [x] Fixed issues caused by changing eslint & CJS
- [x] Formatted everything (Sorry to all the conflicts you get from this)
- [x] Added script to generate jwt (`yarn gen-jwt`) (Resolves #68)
- [x] Fix CI workflow to check instead of write (Resolves #59)
- [x] Add caching to CI workflows (Resolves #67)

In short, mega pr! May the conflicts be with you.

### CODE REVIEW

**Comment by Timothy-Gonzalez** [comment]
I'm sorry in advance for the conflicts this will cause, but it had to be done.

File: src/constants.ts [hunk 1]
```diff
@@ -1,66 +1,74 @@
...
-	private static readonly IOS_REDIRECT:string = "hackillinois://login/";
+    static readonly DEFAULT_DEVICE: string = this.WEB_DEVICE;
+    static readonly DEFAULT_REDIRECT: string = this.WEB_REDIRECT;
 
-	private static readonly ANDROID_DEVICE:string = "android";
-	private static readonly ANDROID_REDIRECT:string = "hackillinois://login/";
-	
-	static readonly DEFAULT_DEVICE:string = this.WEB_DEVICE;
-	static readonly DEFAULT_REDIRECT:string = this.WEB_REDIRECT;
+    static readonly REDIRECT_MAPPINGS: Map<string, string> = new Map<
```
Comment (AydanPirani) [inline src/constants.ts:36]: @Timothy-Gonzalez  What's happening here?

File: src/database.ts [hunk 2]
```diff
@@ -1,35 +1,41 @@
-import { IModelOptions } from "@typegoose/typegoose/lib/types";
-import "dotenv";
+import { IModelOptions } from "@typegoose/typegoose/lib/types.js";
```
Comment (AydanPirani) [inline src/database.ts:1]: Is this expected?

**Review by AydanPirani — COMMENTED** [review]
LGTM for the most part, few questions @Timothy-Gonzalez

**Comment by lasyaneti** [comment]
👍

File: src/database.ts (hunk 2 above)
Comment (Timothy-Gonzalez) [reply src/database.ts:1]: Yes this is from cjs -> esm

File: src/constants.ts (hunk 1 above)
Comment (Timothy-Gonzalez) [reply src/constants.ts:36]: Not the best diff but I think this is just formatting? Not the cleanest diff, but since it compiled everything is probably defined correctly. Formatters arent perfect either but the important thing is consistency. Will check when I get to a computer.

**Review by AydanPirani — APPROVED** [review]
lgtm! merging :)
